### PR TITLE
osx_cc_wrapper: Only expand existing response files

### PR DESCRIPTION
**Pull request by aherrmann**
Closes #13044 
Applies the changes suggested in https://github.com/bazelbuild/bazel/pull/13044#issuecomment-783539568

Not all arguments starting with `@` represent response files, e.g. `-install_name @rpath/...` or `-Xlinker -rpath -Xlinker @loader_path/...` do not refer to response files and attempting to read them as such will fail the build.

Users don't always have control over these arguments, meaning transforming them to `-Wl,-install_name,@rpath/...` or `-Wl,-rpath,@loader_path/...` is not always an option. E.g. as described in https://github.com/bazelbuild/bazel/pull/13044#issuecomment-779380394 where these flags are emitted by the Haskell compiler GHC (see https://github.com/bazelbuild/bazel/pull/13044#issuecomment-782228172 for their reasoning).

With this change `osx_cc_wrapper` will only interpret arguments starting with `@` as response files if the corresponding file exists and is readable. This is analogous to the behavior defined in `wrapped_clang.cc`. See https://github.com/bazelbuild/bazel/pull/13044#issuecomment-783539568 for discussion.

### CODE REVIEW

File: tools/cpp/osx_cc_wrapper.sh [hunk 1]
```diff
@@ -53,7 +53,7 @@ function parse_option() {
 
 # let parse the option list
 for i in "$@"; do
-    if [[ "$i" = @* ]]; then
+    if [[ "$i" = @* && "${i:1}" ]]; then
```
Comment (keith) [inline tools/cpp/osx_cc_wrapper.sh:56]:
```suggestion
    if [[ "$i" = @* && -r "${i:1}" ]]; then
```

Comment (aherrmann) [reply tools/cpp/osx_cc_wrapper.sh]: Thanks!

**Review by keith — APPROVED** [review]

**Review by cocreature — APPROVED** [review]
Thanks for taking this over @aherrmann!

**Comment by keith** [comment]
@trybka mind reviewing this one?

**Review by trybka — APPROVED** [review]
LGTM

**Comment by keith** [comment]
@trybka can you help us land this?

**Comment by trybka** [comment]
On it

**Comment by trybka** [comment]
It's running through CI now. Given that it is Friday afternoon here, I don't expect this to land before Monday.

**Pull request closed without merging**